### PR TITLE
Model HMG-50 input-conditioning gate (spike filter, deadband)

### PR DESCRIPTION
**Pull request by tomquist**
Extends the firmware steering controller to model the real HMG-50's input-conditioning gate that runs before the ramp law. This gate implements three hold conditions:

- **>50 W spike filter** — grid jumps over 50 W that the battery's own output change (<20 W) cannot explain are treated as transients and skipped for one sample. The filter has no one-shot: a sustained drift whose own output never moves keeps being skipped.
- **±20 W deadband** — `abs(g) < 20` with the battery's own output below 1 W parks the setpoint. The test is signed: a charging battery (negative output) is also held.
- **Small-import hold** — a residual import of `0 <= g < 10` is held even while the battery is producing, so it doesn't chase the last few watts.

## Key changes

- **`firmware_steering.py`**: 
  - Renamed `step()` to `step_raw()` to expose the bare ramp law
  - Added new `step()` method that runs the input gate before calling `step_raw()`
  - Implemented `_gate()` method with the three hold conditions
  - Added gate state tracking (`prev_g`, `prev_out`) that advances on every cycle, including held samples
  - Extracted share-split logic into `_share_split()` helper
  - Updated module docstring to clarify this models the **HMG-50** specifically (not VNSE3-0)
  - Exported gate threshold constants (`DEADBAND_W`, `SPIKE_JUMP_W`, `SPIKE_OWN_DELTA_W`, `SMALL_IMPORT_HOLD_W`)

- **`firmware_steering_test.py`**:
  - Updated `GOLDEN` test data to use `step_raw()` (bare ramp law only)
  - Added `GATED` reference trajectories covering the full `step()` pipeline with 6 scenarios
  - Added `test_matches_gated_golden_trajectory()` to lock the gate behavior
  - Added 10 unit tests covering gate boundaries and edge cases (deadband, spike filter, small-import hold, etc.)
  - Updated module docstring to document both `GOLDEN` and `GATED` test sets

- **`battery.py`**:
  - Removed `idle_on_cross_phase_discharge` flag (the real firmware does not idle on cross-phase discharge signals)
  - Updated `_handle_ct_response()` to pass `out=self._current_power` to the steering controller, enabling the gate's own-output checks
  - Updated docstring to clarify the gate's role

- **`runner.py`**: Removed `idle_on_cross_phase_discharge` from `BatteryConfig`

- **`battery_test.py`**: Replaced three tests for the removed idle flag with three new tests validating gate behavior (deadband, spike debouncing, own-output dependency)

- **`ct002-ct003-protocol.md`**: Updated documentation to clarify the HMG-50 scope, document the gate's three conditions and their ordering, and note that VNSE3-0 runs a different steering law

- **`test_efficiency_e2e.py` and `test_e2e_probe_lockup.py`**: Relaxed assertions to account for the gate's debouncing and deadband introducing transient grid errors during probe startup and handoff

- **`test_issue376_pv_passthrough.py`**: Removed `idle_on_cross_phase_discharge` from battery configs

- **`CHANGELOG.md`**: Updated to reflect the gate implementation and removal of the cross-phase idle flag

https://claude.ai/code/session_01ErMiuN6o5qWwXsSz9pptGy

## Summary by CodeRabbit

* **New Features**
  * DC-coupled B2500 battery simulation with per-channel integer steering and meter-type selection.

* **Bug Fixes**
  * Steering logic updated to mirror real device behavior: input conditioning (spike filter, deadband, small-import hold), fair per-phase sharing, and honoring participation flags; removed non-physical idle-on-cross-phase option.

* **Documentation**
  * Expanded protocol docs with detailed steering, gating, and B2500 DC-output behavior.

* **Tests**
  * Added comprehensive unit and end-to-end tests for steering and mixed-device scenarios.

### CODE REVIEW

File: src/astrameter/simulator/runner.py [hunk 1]
```diff
@@ -175,6 +175,7 @@ def parse_config(data: dict) -> SimulationConfig:
         bc = BatteryConfig(
             mac=bd["mac"],
             phase=bd["phase"],
+            meter_dev_type=str(bd.get("meter_dev_type", "HMG-50")),
```
[the remark this answers is not in the document]
Comment (tomquist) [reply src/astrameter/simulator/runner.py]: Partially applied in e8a9144: `parse_config` now rejects non-string `meter_dev_type` values (so e.g. `null` no longer coerces to `"None"` and silently classifies as a Venus), with tests.

The allowlist/normalization half is intentionally not applied: any *string* is a valid device type by design — `device_capabilities` documents that unknown/future/empty model strings classify as modern AC-coupled batteries, mirroring how the real CT treats device types it doesn't know. Validating against a closed set would break simulating new models, and `device_capabilities` never raises, so the suggested `try/except` in `validate_config` would be dead code. Upper-casing at parse time is also unnecessary (classification already upper-cases internally) and would change the verbatim device-type string the simulator echoes on the wire.

---
_Generated by [Claude Code](https://claude.ai/code)_
